amazon-kinesis-client/src/test/java/software/amazon/kinesis/common/StreamARNUtilTest.java

```diff
@@ -0,0 +1,70 @@
+import org.powermock.modules.junit4.PowerMockRunner;
+import software.amazon.awssdk.arns.Arn;
+import software.amazon.awssdk.regions.Region;
+
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ StreamARNUtil.class })
+public class StreamARNUtilTest {
+    private static final String STS_RESPONSE_ARN_FORMAT = "arn:aws:sts::%s:assumed-role/Admin/alias";
+    private static final String KINESIS_STREAM_ARN_FORMAT = "arn:aws:kinesis:us-east-1:%s:stream/%s";
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        PowerMockito.spy(StreamARNUtil.class);
+    }
+
+    @Test
+    public void testGetStreamARNHappyCase() throws Exception {
+        String streamName = "testStreamName";
+        String accountId = "123456789012";
+        PowerMockito.doReturn(Optional.of(Arn.fromString(String.format(STS_RESPONSE_ARN_FORMAT, accountId))))
+                .when(StreamARNUtil.class, "getStsCallerArn");
+
+        Optional<Arn> actualStreamARNOptional = StreamARNUtil.getStreamARN(streamName, Region.US_EAST_1);
+        String expectedStreamARN = String.format(KINESIS_STREAM_ARN_FORMAT, accountId, streamName);
+        assertTrue(actualStreamARNOptional.isPresent());
+        assertEquals(expectedStreamARN, actualStreamARNOptional.get().toString());
+    }
+
+    @Test
+    public void testGetStreamARNReturnsEmptyOnSTSError() throws Exception {
+        // Optional.empty() is expected when there is an error with the STS call and STS returns empty Arn
+        PowerMockito.doReturn(Optional.empty()).when(StreamARNUtil.class, "getStsCallerArn");
+        Optional<Arn> actualStreamARNOptional = StreamARNUtil.getStreamARN("testStreamName", Region.US_EAST_1);
+        assertFalse(actualStreamARNOptional.isPresent());
+    }
+
+    @Test
+    public void testGetStreamARNReturnsEmptyOnInvalidKinesisRegion() {
+        // Optional.empty() is expected when kinesis region is not set correctly
+        Optional<Arn> actualStreamARNOptional = StreamARNUtil.getStreamARN("testStreamName", null);
+        assertFalse(actualStreamARNOptional.isPresent());
+    }
+
+    @Test
+    public void testGetStreamARNWithProvidedAccountIDAndIgnoredSTSResult() throws Exception {
+        // If the account id is provided in the StreamIdentifier, it will override the result (account id) returned by sts
+        PowerMockito.doReturn(Optional.of(Arn.fromString(String.format(STS_RESPONSE_ARN_FORMAT, "111111111111"))))
+                .when(StreamARNUtil.class, "getStsCallerArn");
+        Optional<Arn> actualStreamARNOptional = StreamARNUtil.getStreamARN(
+                "testStreamName", Region.US_EAST_1, Optional.of("222222222222"));
+        assertTrue(actualStreamARNOptional.isPresent());
+        assertTrue(actualStreamARNOptional.get().toString().contains("222222222222"));
```

Comment:
Would you kindly restructure this as an `assertEquals(<expected>, <actual>)`? If/When this assertion fails, all the engineer will know is that it wasn't `true`. `assertEquals(...)` provides a comparison which greatly facilitates debugging.